roles/tests-integration/tests/sniffer_integration.rs

```diff
@@ -152,3 +152,44 @@ async fn test_sniffer_wait_for_message_type_with_remove() {
         false
     );
 }
+
+/// Verifies that [`Sniffer`] can intercept and block a message stream.
+///
+/// This test sets up a chain where a message from the Template Provider (TP)  
+/// passes through two sniffers (`sniffer_a` and `sniffer_b`) before reaching the Pool.
+///
+/// - `sniffer_a` is configured to block `SetupConnectionSuccess` messages directed downstream.
+/// - `sniffer_b` should receive no messages after initial setup, ensuring the block works.
+///
+/// **Flow:**  
+/// `TP -> sniffer_a -> sniffer_b -> Pool`  
+#[tokio::test]
+async fn test_sniffer_blocks_message() {
+    start_tracing();
+    let (_tp, tp_addr) = start_template_provider(None);
+
+    // Define an action to block SetupConnectionSuccess messages going downstream.
+    let block_from_message = BlockFromMessage::new(
+        MessageDirection::ToDownstream,
+        MESSAGE_TYPE_SETUP_CONNECTION_SUCCESS,
+    );
+
+    // `sniffer_a` intercepts and blocks `SetupConnectionSuccess` messages.
+    let (_sniffer_a, sniffer_a_addr) = start_sniffer(
+        "A".to_string(),
+        tp_addr,
+        false,
+        Some(sniffer::Action::BlockFromMessage(block_from_message)),
```

Comment:
It would be nice to implement `impl From<BlockFromMessage> for Action` and `From<InterceptMessage>`, this way we can just `block_from_message.into()`.